### ISSUE TITLE
Support provider context

### ISSUE THREAD

**Issue by aweiteka** [projectatomic/atomicapp#75]
Application context and namespacing is a key feature of providers such as kubernetes. Here's how I think it should work.

### Definition

Default namespace is defined in the nulecule file in the metadata section.

```
metadata:
  kubernetes:
    context: mycluster
    namespace: mycoolapp
...
```

At deployment the default context and namespace maybe overridden using a general parameter.

```
[general]
provider=kubernetes
context=aweiteka-cluster
namespace=mycoolerapp
...
```

### Provider implementation

If provided the provider plugin should create resources with the context and/or namespace options.

```
kubectl create -f mycoolapp-pod.json --context=aweiteka-cluster --namespace=mycoolapp
```

The context and namespace must exist for the above create command to succeed. It appears to be safe to do this even if values exist.
```
kubectl config set-context aweiteka-cluster --namespace=mycoolapp
```

#### Configuration file

It appears `~/.kube/.kubeconfig` file does not exist by default and causes `set-context` calls to fail. Merely touching this file resolves the failure.

```
$ kubectl config set-context myapp --namespace=mycoolapp
open /home/vagrant/.kube/.kubeconfig: no such file or directory
$ sudo kubectl config set-context myapp --namespace=mycoolapp
open /root/.kube/.kubeconfig: no such file or directory
$ mkdir /home/vagrant/.kube/
$ touch /home/vagrant/.kube/.kubeconfig
```

#### Viewing and set the context namespace 
```
$ kubectl config view
apiVersion: v1
clusters: []
contexts: []
current-context: ""
kind: Config
preferences: {}
users: []
$ kubectl config set-context myapp --namespace=mycoolapp
$ kubectl config view
apiVersion: v1
clusters: []
contexts:
- context:
    cluster: ""
    namespace: mycoolapp
    user: ""
  name: myapp
current-context: ""
kind: Config
preferences: {}
users: []
```

**Comment by aweiteka** [comment]
@goern is this a dupe of #60  ?

**Comment by goern** [comment]
@aweiteka didnt we do that via namespaces?

**Comment by aweiteka** [comment]
@vpavlin please reference your provider config work.

**Comment by vpavlin** [comment]
Closed by #267
